### ISSUE TITLE
Programmatically calling search results

### ISSUE THREAD

**Issue by kephale** [imagej/napari-imagej#241]
I was hoping to programmatically make calls to commands/scripts with napari-imagej.

I've read up to `ResultRunner`, `jc.SearchResult`, and `python_actions_for` but it looks like search is pretty directly connected to the UI. 

Is there a way to do something like:
```
endpoint = "sc.fiji:fiji:2.13.0+org.morphonets:SNT:MANAGED"

...

search_results = get_matching_commands("tracing")
command_result = run_command(search_results[0], <some args>)
```

**Comment by gselzer** [comment]
@kephale you can get much of this functionality from pyimagej, but there's of course additional value in the processing napari-imagej adds on top of pyimagej. And @ctrueden asks for a lot of this functionality in #208 - the functionality is, like you say, too tied in to the GUI. A better separation of functionality from UI is already on the docket!

What's your use case here? I really want to support stuff like this - I'm just not sure whether it is napari-imagej's job or pyimagej's job to do it.

**Comment by kephale** [comment]
I'm trying to port workflows from purely being run within Fiji to being run as Python scripts, so I'm starting with a skeleton script that sets up an endpoint, then incrementally adding/tweaking code line-by-line.

I think the main thing I want is to have Python functions that correspond to IJ2 `Command`s which have the logic for taking napari layers as input. It is definitely possible to do this stuff with pyimagej, but it seems as though napari-imagej is already providing the logic for mapping from IJ2 types to napari layers. 

More on using pyimagej+napari-imagej in another issue.

**Comment by ctrueden** [comment]
@gselzer Is it possible to make use of napari-imagej's layer conversion logic programmatically by using `ij.py.to/from_java`, since napari-imagej registers that conversion logic as scyjava converters?

**Comment by gselzer** [comment]
> @gselzer Is it possible to make use of napari-imagej's layer conversion logic programmatically by using `ij.py.to/from_java`, since napari-imagej registers that conversion logic as scyjava converters?

Yup, all you have to do is register the converters (either by letting napari-imagej install them for you by calling `init_ij`, or manually by calling [`install_converters`](https://github.com/imagej/napari-imagej/blob/7d88ee154a4f009a028dea65d30e9053a93fc7f1/src/napari_imagej/types/converters/__init__.py#L92))

**Comment by ctrueden** [comment]
OK, so @kephale's script could use `napari_imagej.init_ij` to synchronously initialize PyImageJ according to its current settings, and then he's good to go? I guess that still doesn't resolve this issue though, which is about exposing programmatic access to the search functions. You could do that now by using the Java API by accessing the `SearchService`, but there's nothing more Pythonic yet. What else (if anything) does napari-imagej add on top of the search service that would be good to expose?

**Comment by ctrueden** [comment]
As part of my work on #265, I'm creating a `model.py` with `NapariImageJ` central class holding all the non-Qt-specific business logic. This will include the REPL of course, but I'll also see about making the search logic part of that as well. Then maybe this issue can be closed out at the same time as #265, assuming my plan is successful. API calls for everyone! :birthday:
